crates/s3s/src/signature_v4/methods.rs

```diff
@@ -137,7 +138,7 @@ pub fn create_canonical_request(
 
     {
         // <CanonicalURI>\n
-        uri_encode(&mut ans, uri_path, false);
+        ans.push_str(uri_path);
```

Comment:
[AWS signature v4](https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html) defines their custom uri encoding function. The uri path from `hyper` may use different encoding. So the problem is not here.

We should decode the uri path and then pass it to the signature functions.